dlib/serialize.h

```diff
@@ -1773,6 +1773,76 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template<typename T>
+    inline void serialize_these(std::ostream& out, const T& x)
+    {
+        using dlib::serialize;
+        serialize(x, out);
+    }
+    
+    template<typename T, typename... Rest>
+    inline void serialize_these(std::ostream& out, const T& x, const Rest& ... rest)
+    {
+        serialize_these(out, x);
+        serialize_these(out, rest...);
+    }
+    
+    template<typename T>
+    inline void deserialize_these(std::istream& in, T& x)
+    {
+        using dlib::deserialize;
+        deserialize(x, in);
+    }
+    
+    template<typename T, typename... Rest>
+    inline void deserialize_these(std::istream& in, T& x, Rest& ... rest)
+    {
+        deserialize_these(in, x);
+        deserialize_these(in, rest...);
+    }  
+    
+    #define DLIB_DEFINE_DEFAULT_SERIALIZATION(Type, ...)    \
+    void serialize_all(std::ostream& out) const             \
```

Comment:
This all looks good.  How about renaming `serialize_all` and `deserialize_all` to `serialize_to` and `deserialize_from`.  That way it will read nice for any users who happen to write `my_object.serialize_to(out);` or `my_object.deserialize_from(in)`;